### ISSUE TITLE
RSA signatures with pre-calculated hash, but digest set are calculated wrong

### ISSUE THREAD

**Issue by fabled** [latchset/pkcs11-provider#113]
**Describe the bug**
Doing RSA PKCS1.5 signature with pre-calculated hash with hash information is calculated wrong. A DigestInfo struct should be included in the data to be signed.

**To Reproduce**
Patch to add for the test suite:
``` diff
diff --git a/tests/tbasic b/tests/tbasic
index 7153cb3..07015e9 100755
--- a/tests/tbasic
+++ b/tests/tbasic
@@ -54,6 +54,19 @@ pkeyutl -verify -inkey "${PUBURI}"
                 -in ${TMPPDIR}/sha256.bin
                 -sigfile ${TMPPDIR}/sha256-sig.bin'
 
+title PARA "Sign and Verify with provided Hash and RSA with DigestInfo struct"
+ossl 'dgst -sha256 -binary -out ${TMPPDIR}/sha256.bin ${SEEDFILE}'
+ossl '
+pkeyutl -sign -inkey "${PRIURI}" -pkeyopt digest:sha256
+              -in ${TMPPDIR}/sha256.bin
+              -out ${TMPPDIR}/sha256-sig.bin'
+
+ossl '
+pkeyutl -verify -inkey "${PUBURI}" -pkeyopt digest:sha256
+                -pubin
+                -in ${TMPPDIR}/sha256.bin
+                -sigfile ${TMPPDIR}/sha256-sig.bin'
+
 title PARA "Sign and Verify with provided Hash and EC"
 ossl '
 pkeyutl -sign -inkey "${ECBASEURI}"
```

1. Apply above patch
2. `make check`

**Expected behavior**
Modified test suite passes.

**Additional context**
The OpenSSL core will properly set rsasig context, and result in log:
> Set OSSL_SIGNATURE_PARAM_DIGEST to sha256

However, the sequence to call sign/verify will end up in a call to `p11prov_sig_set_mechanism(digest_sign=false)` and the digest gets completely ignored there.

It seems PKCS11 does not support signature with any of the `CKM_SHA256_RSA_PKCS` etc., so it is likely that the provider code should calculate and produce the `DigestInfo` structure with the hash info and use `CKM_RSA_PKCS` sign/verify mechanism.

**Comment by fabled** [comment]
@simo5 I have the fix commit now in my repo. It builds on top of the other two PRs, so I'll create a PR once those are handled first. You may want to take a look at the fix commit already to see if it looks ok, and how it leverages the other two PRs...

**Comment by simo5** [comment]
This looks good, if you can fix DER_NULL handling in #117 we can quickly add this one too.
Ignore the request for tests there, I see you need this piece of code for them to pass.
